Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -57,222 +56,277 @@ def test_test_module(requests_mock, mock_client):
-    with pytest.raises((ValueError, IllumioException, InvalidPortException, InvalidSingleSelectException,
-                        InvalidMultiSelectException)) as err:
-        illumio_traffic_analysis_command(mock_client, args)
-    assert str(err.value) == err_msg
+    from IllumioCore import traffic_analysis_command
+
+    with pytest.raises(err_type) as err:
+        traffic_analysis_command(mock_client, args)
+        assert str(err.value) == err_msg
 
 
-def test_illumio_traffic_analysis_success(requests_mock, mock_client):
+def test_traffic_analysis_success(requests_mock, mock_client):
     """
-    Test case scenario for successful execution of illumio_traffic_analysis_command function.
+    Test case scenario for successful execution of traffic-analysis-command function.
 
     Given:
-        - command arguments for illumio_traffic_analysis
+        - command arguments for traffic_analysis
     When:
-        - Calling `illumio_traffic_analysis_Command` function
+        - Calling `traffic_analysis_Command` function
     Then:
-        - Returns a valid output
+        - Returns a valid raw_response.
     """
-    from IllumioCore import illumio_traffic_analysis_command
+    args = {
+        "port": 8443,
+        "protocol": "tcp",
+        "policy_decisions": "potentially_blocked",
+        "start_time": "2022-07-17T12:58:33.528Z",
+        "end_time": "2022-07-18T12:58:33.529Z",
+    }
 
-    args = {'port': 8443, 'protocol': "tcp", 'policy_decisions': 'potentially_blocked',
-            'start_time': "2022-07-17T12:58:33.528Z", 'end_time': "2022-07-18T12:58:33.529Z"}
+    json_data = util_load_json(os.path.join(DIRECTORY_PATH, "test_data/traffic_analysis_success_response.json"))
 
-    json_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                            "test_data/traffic_analysis_success_response.json"))
+    json_data_get = util_load_json(os.path.join(DIRECTORY_PATH, "test_data/traffic_analysis_get_response.json"))
 
-    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                           "./test_data/traffic_analysis_success_hr.md")) as file:
-        hr_output = file.read()
-
-    json_data_get = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                                "test_data/traffic_analysis_get_response.json"))
-
-    json_download_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                                     "test_data/traffic_analysis_download_data.json"))
+    json_download_data = util_load_json(os.path.join(DIRECTORY_PATH, "test_data/traffic_analysis_download_data.json"))
 
-    requests_mock.post("https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries",
-                       json=json_data)
+    requests_mock.post("https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries", json=json_data)
```

Comment:
You only need the path to match for the requests_mock call:

```
    pattern = re.compile('/traffic_flows/async_queries')
    requests_mock.get(pattern, json=json_data)
```

This is more future-proof (updated API versions, testing using different org ID).

Reply:
See my other comment in the longer thread though, this isn't a great way to test this function.

Reply:
Yes, as we're using monkey-patching this issue is resolved, in-case if we need path will use `re.compile` instead.